src/engine/renderer/tr_image.cpp

```diff
@@ -796,6 +785,81 @@ R_ConvertBC5Image(const byte **in, byte **out, int numMips, int numLayers,
 	}
 }
 
+int CheckImage3D( GLint internalFormat, GLsizei width, GLsizei height, GLint border, GLenum format, GLenum type )
+{
+	/* From https://www.khronos.org/opengl/wiki/GLAPI/glTexImage3D
+
+	> If target is GL_PROXY_TEXTURE_3D, no data is read from data,
+	> but all of the texture image state is recalculated, checked
+	> for consistency, and checked against the implementation's
+	> capabilities. If the implementation cannot handle a texture
+	> of the requested texture size, it sets all of the image state
+	> to 0, but does not generate an error (see glGetError).
+	> To query for an entire mipmap array, use an image array level
+	> greater than or equal to 1.
+
+	From https://www.khronos.org/opengl/wiki/GLAPI/glTexImage3D
+
+	> To then query this state, call glGetTexLevelParameter. */
+
+	glTexImage3D( GL_PROXY_TEXTURE_3D, 0, internalFormat, width, height, 0, border, format, type, nullptr );
+
+	GLint finalWidth;
+	glGetTexLevelParameteriv( GL_PROXY_TEXTURE_3D, 0, GL_TEXTURE_WIDTH, &finalWidth );
+
+	if ( finalWidth == 0 )
+	{
+		width >>= 1;
+		height >>= 1;
+		return 1 + CheckImage3D( internalFormat, width, height, border, format, type );
+	}
+
+	return 0;
+}
+
+int CheckImageCubeMap( GLint internalFormat, GLsizei width, GLsizei height, GLint border, GLenum format, GLenum type )
+{
+	/* It have been noticed ATI X1950 PRO reports 4096 max cube map size
+	and it does not work with cubemaps having 1024 sized edge.
+	It was also noticed than when multiplying width and height by at least 5,
+	faces get downscaled one time to 512 and they work.
+
+	At this point there is no explanation for the need of such multiplier,
+	and if there is one, what is the multiplier to use,
+	so let's just use the amount of faces, 6. */
+
+	glTexImage2D( GL_PROXY_TEXTURE_CUBE_MAP, 0, internalFormat, width * 6, height * 6, border, format, type, nullptr );
```

Comment:
This calculation doesn't seem right, since if you multiply both the width and height by 6, then the image uses 36 times as much memory.

But in any case it doesn't seem good to add a hack like this, since it could cause images to be wrongly downscaled on hardware that actually supports it.

Reply:
Deleted the bizarre `6`. Also, I noticed I was doing the check before internalFormat is determined (hence always using the same default one).